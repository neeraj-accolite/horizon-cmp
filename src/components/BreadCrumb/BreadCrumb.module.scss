/* BreadCrumb.module.scss */
.breadcrumb {
  display: flex; /* Arrange items inline */
  align-items: center; /* Vertically center items */
  font-size: 14px;
  color: #555; /* Default text color */
}
