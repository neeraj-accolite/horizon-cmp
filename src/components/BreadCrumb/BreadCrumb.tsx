import React from 'react';
import { Breadcrumbs, BreadcrumbItem } from '@nextui-org/react';

export interface BreadCrumbProps {
  items: { label: string; href?: string }[];
}

const BreadCrumb: React.FC<BreadCrumbProps> = ({ items }) => {
  const [currentPage, setCurrentPage] = React.useState(
    items.length > 0 ? items[items.length - 1].label : '',
  );

  if (items.length === 0) {
    return null; // Render nothing if there are no items
  }

  const handleClick = (itemLabel: string, href?: string) => {
    setCurrentPage(itemLabel);
    if (href) {
      window.location.href = href; // Navigate to the provided URL
    }
  };
  return (
    <Breadcrumbs>
      {items.map((item) => (
        <BreadcrumbItem
          onPress={() => handleClick(item.label, item.href)}
          key={item.label}
          href={item.href}
          isCurrent={currentPage === item.label}
          style={{ fontWeight: currentPage === item.label ? 'bold' : 'normal' }}
        >
          {item.label}
        </BreadcrumbItem>
      ))}
    </Breadcrumbs>
  );
};

export default BreadCrumb;
