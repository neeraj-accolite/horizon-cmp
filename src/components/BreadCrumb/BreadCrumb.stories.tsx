import React from 'react';
import { Meta, StoryFn } from '@storybook/react';
import BreadCrumb, { BreadCrumbProps } from './BreadCrumb'; // Correct path

export default {
  title: 'Components/BreadCrumb', // This is required
  component: BreadCrumb, // Link the component to the stories
} as Meta;

const Template: StoryFn<BreadCrumbProps> = (args) => <BreadCrumb {...args} />;

export const Default = Template.bind({});
Default.args = {
  items: [
    { label: 'Home', href: '/' },
    { label: 'Services', href: '/services' },
    { label: 'Plumbing', href: '/plumbing' },
  ],
};

export const WithoutLinks = Template.bind({});
WithoutLinks.args = {
  items: [{ label: 'Home' }, { label: 'Plumbing' }],
};
