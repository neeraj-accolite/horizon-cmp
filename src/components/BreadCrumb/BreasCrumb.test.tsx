import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import BreadCrumb from './BreadCrumb';
import { BreadCrumbProps } from './BreadCrumb';

describe('BreadCrumb Component', () => {
  const renderComponent = (props: BreadCrumbProps) =>
    render(<BreadCrumb {...props} />);

  test('renders all breadcrumb items with links', () => {
    const items = [
      { label: 'Home', href: '/' },
      { label: 'Services', href: '/services' },
      { label: 'Plumbing', href: '/services/plumbing' },
    ];

    renderComponent({ items });

    items.forEach((item) => {
      expect(screen.getByText(item.label)).toBeInTheDocument();
    });

    const separators = screen.getAllByText('>');
    expect(separators.length).toBe(items.length - 1); // One less than the number of items
  });

  test('renders all breadcrumb items without links', () => {
    const items = [
      { label: 'Home' },
      { label: 'Services' },
      { label: 'Plumbing' },
    ];

    renderComponent({ items });

    // Check all breadcrumb items are rendered
    items.forEach((item) => {
      const element = screen.getByText(item.label);
      expect(element).toBeInTheDocument();
      expect(element.tagName).toBe('DIV');
    });

    // Check separator (>)
    const separators = screen.getAllByText('>');
    expect(separators.length).toBe(items.length - 1); // One less than the number of items
  });

  test('highlights the last breadcrumb item', () => {
    const items = [
      { label: 'Home', href: '/' },
      { label: 'Services', href: '/services' },
      { label: 'Plumbing' },
    ];

    renderComponent({ items });

    const lastItem = screen.getByText('Plumbing');
    expect(lastItem).toBeInTheDocument();
    expect(lastItem).toHaveClass('text-gray-800'); // Verify the last item is highlighted
    expect(lastItem).toHaveClass('font-bold');
  });

  test('renders correctly with an empty items array', () => {
    const { container } = render(<BreadCrumb items={[]} />);

    // Check that the container for breadcrumbs is empty
    const breadcrumbContainer = container.firstChild;
    expect(breadcrumbContainer).toBeEmptyDOMElement();
  });

  test('handles edge case with only one item', () => {
    const items = [{ label: 'Home', href: '/' }];

    renderComponent({ items });

    // Check that the item is rendered without a separator
    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.queryByText('>')).not.toBeInTheDocument();
  });

  test('matches the snapshot', () => {
    const items = [
      { label: 'Home', href: '/' },
      { label: 'Services', href: '/services' },
      { label: 'Plumbing' },
    ];

    const { asFragment } = render(<BreadCrumb items={items} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
