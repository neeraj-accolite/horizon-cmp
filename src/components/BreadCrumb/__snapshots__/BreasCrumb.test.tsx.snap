// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumb Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center text-gray-600 text-sm"
  >
    <div
      class="flex flex-row items-center"
    >
      <a
        class="no-underline cursor-pointer"
        href="/"
      >
        Home
      </a>
      <div
        class="mx-2 text-gray-800"
      >
        &gt;
      </div>
    </div>
    <div
      class="flex flex-row items-center"
    >
      <a
        class="no-underline cursor-pointer"
        href="/services"
      >
        Services
      </a>
      <div
        class="mx-2 text-gray-800"
      >
        &gt;
      </div>
    </div>
    <div
      class="flex flex-row items-center"
    >
      <div
        class="font-bold text-gray-800"
      >
        Plumbing
      </div>
    </div>
  </div>
</DocumentFragment>
`;
